internal/cmd/login.go

```diff
@@ -154,7 +154,8 @@ func login(options loginCmdOptions) error {
 func loginToInfra(client *api.Client, loginReq *api.LoginRequest) error {
 	loginRes, err := client.Login(loginReq)
 	if err != nil {
-		if errors.Is(err, api.ErrUnauthorized) {
+		var apiError api.Error
+		if errors.As(err, &apiError) && apiError.Code == http.StatusUnauthorized {
```

Comment:
If we want to make this operation more convenient, I think we could add this function to the `api` package. `grpc` has a similar helper that makes it convenient to deal with their status codes.

```go
func ErrorStatusCode(err error) (int32, bool) {
    var apiError api.Error
    if errors.As(err, &apiError) {
        return apiError.Code, true
    }
    return 0, false
}
```